components/common/src/command/package/install.rs

```diff
@@ -672,11 +672,13 @@ impl<'a> InstallTask<'a> {
                    ident);
         } else if self.is_offline() {
             return Err(Error::OfflineArtifactNotFound(ident.as_ref().clone()));
-        } else if retry(delay::Fixed::from(RETRY_WAIT).take(RETRIES), fetch_artifact).is_err() {
-            return Err(Error::DownloadFailed(format!("We tried {} times but \
-                                                      could not download {}. \
-                                                      Giving up.",
-                                                     RETRIES, ident)));
+        } else {
```

Comment:
Why not
```rust
        } else if let Err(err) = retry(…) {
            return Err(…);
        }
```
?

Reply:
oh i thought you could not do `else if let`. that is better